sycl/doc/extensions/proposed/sycl_ext_oneapi_device_is_integrated.asciidoc

```diff
@@ -0,0 +1,121 @@
+= sycl_ext_oneapi_device_is_integrated
+
+:source-highlighter: coderay
+:coderay-linenums-mode: table
+
+// This section needs to be after the document title.
+:doctype: book
+:toc2:
+:toc: left
+:encoding: utf-8
+:lang: en
+:dpcpp: pass:[DPC++]
+:endnote: &#8212;{nbsp}end{nbsp}note
+
+// Set the default source code type in this document to C++,
+// for syntax highlighting purposes.  This is needed because
+// docbook uses c++ and html5 uses cpp.
+:language: {basebackend@docbook:c++:cpp}
+
+
+== Notice
+
+[%hardbreaks]
+Copyright (C) 2025 Intel Corporation.  All rights reserved.
+
+Khronos(R) is a registered trademark and SYCL(TM) and SPIR(TM) are trademarks
+of The Khronos Group Inc.  OpenCL(TM) is a trademark of Apple Inc. used by
+permission by Khronos.
+
+
+== Contact
+
+To report problems with this extension, please open a new issue at:
+
+https://github.com/intel/llvm/issues
+
+
+== Dependencies
+
+This extension is written against the SYCL 2020 revision 10 specification.
+All references below to the "core SYCL specification" or to section numbers in
+the SYCL specification refer to that revision.
+
+
+== Status
+
+This is a proposed extension specification, intended to gather community
+feedback.  Interfaces defined in this specification may not be implemented yet
+or may be in a preliminary state.  The specification itself may also change in
+incompatible ways before it is finalized.  *Shipping software products should
+not rely on APIs defined in this specification.*
+
+
+== Overview
+
+This extension allows host code to check whether a physical device, represented
+by a SYCL device, is integrated with the host.
```

Comment:
I'd suggest a slight rewording here because the difference between "integrated" and "discrete" isn't always clear cut:

> This extension allows host code to check whether a SYCL GPU device is "integrated" or "discrete".  An integrated GPU device often shares memory with the host CPU, while a discrete GPU is often a separate card with its own memory.  However, some GPU devices may not fit neatly into either category, so implementors may provide their own judgement when categorizing a GPU device as "integrated" vs. "discrete".

@bashbaug does this seem like a reasonable description?  Any suggestions?